src/dns.mli

```diff
@@ -341,19 +350,72 @@ module Dnskey : sig
   (** [pp ppf t] pretty-prints the DNSKEY. *)
 
   val compare : t -> t -> int
-  (** [comapre a b] compares the DNSKEY [a] with [b]. *)
+  (** [compare a b] compares the DNSKEY [a] with [b]. *)
 
   val of_string : string -> (t, [> `Msg of string ]) result
   (** [of_string str] attempts to parse [str] to a dnskey. The colon character
-      ([:]) is used as separator, supported formats are: [algo:keydata] and
-      [flags:algo:keydata], where keydata is a base64 string. *)
+      ([:]) is used as separator, supported format is: [algo:keydata] where
+      keydata is a base64 string. *)
 
   val name_key_of_string : string -> ([ `raw ] Domain_name.t * t, [> `Msg of string ]) result
   (** [name_key_of_string str] attempts to parse [str] to a domain name and a
       dnskey. The colon character ([:]) is used as separator. *)
 
   val pp_name_key : ([ `raw ] Domain_name.t * t) Fmt.t
   (** [pp_name_key (name, key)] pretty-prints the dnskey and name pair. *)
+
+  val digest_prep : [ `raw ] Domain_name.t -> t -> Cstruct.t
+end
+
+(** RRSIG *)
+module Rrsig : sig
+
+  type t = {
+    type_covered : int ;
+    algorithm : Dnskey.algorithm ;
+    label_count : int ;
+    original_ttl : int32 ;
+    signature_expiration : Ptime.t ;
+    signature_inception : Ptime.t ;
+    key_tag : int ;
+    signer_name : [ `raw ] Domain_name.t ;
+    signature : Cstruct.t
+  }
+  (** The type of a RRSIG. *)
+
+  val pp : t Fmt.t
+  (** [pp ppf t] pretty-prints the RRSIG. *)
+
+  val compare : t -> t -> int
+  (** [compare a b] compares the RRSIG [a] with [b]. *)
+end
+
+(** DS *)
+module Ds : sig
+  type digest_type =
+    | SHA1
+    | SHA256
+    | SHA384
```

Comment:
is this missing GOST from RFC5933? I think that'll cover all the assigned ones

Reply:
indeed (https://www.iana.org/assignments/ds-rr-types/ds-rr-types.xhtml) -- but my approach is to name only these we actually have support for (and all others for future compatibility fall into the `Unknown`). since I don't see a GOST implementation (neither do I know anyone who asked for it), I'll skip that for now.

Reply:
Sounds good. My understanding is that GOST is on the way out anyway.